packages/amplify_storage_s3/android/src/main/kotlin/com/amazonaws/amplify/amplify_storage_s3/AmplifyStorageOperations.kt

```diff
@@ -197,16 +184,23 @@ class AmplifyStorageOperations {
             }
         }
 
-        private fun prepareError(@NonNull flutterResult: MethodChannel.Result, @NonNull msg: String, localizedError: String?, recoverySuggestion: String?) {
-            var errorDetails = HashMap<String, Any>();
-            errorDetails.put("PLATFORM_EXCEPTIONS", mapOf(
-                    "platform" to "Android",
-                    "localizedErrorMessage" to localizedError,
-                    "recoverySuggestion" to recoverySuggestion
-            ))
+        private fun prepareError(@NonNull flutterResult: MethodChannel.Result, @NonNull e: Exception) {
+            val errorCode = "StorageException"
+            LOG.error(errorCode, e)
+            val serializedError: Map<String, Any?> = if (e is AmplifyException) {
+                ExceptionUtil.createSerializedError(e)
+            } else {
+                ExceptionUtil.createSerializedError(
+                        "An unexpected error has occurred",
+                        "An unexpected error has occurred. See Android logs for details",
+                        e.toString())
+            }
             Handler(Looper.getMainLooper()).post {
-                flutterResult.error("AmplifyException", msg, errorDetails)
+                ExceptionUtil.postExceptionToFlutterChannel(flutterResult,
+                        "StorageException",
```

Comment:
You can probably use the variable `errorCode` you defined earlier here

Reply:
Done.